cocos/gfx/webgpu/webgpu-command-allocator.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ Copyright (c) 2024 Xiamen Yaji Software Co., Ltd.
+
+ https://www.cocos.com/
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights to
+ use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ of the Software, and to permit persons to whom the Software is furnished to do so,
+ subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+*/
+
+import { CachedArray } from '../../core/memop/cached-array';
+import {
+    WebGPUCmdBeginRenderPass,
+    WebGPUCmdBindStates,
+    WebGPUCmdCopyBufferToTexture,
+    WebGPUCmdDraw,
+    WebGPUCmdObject,
+    WebGPUCmdPackage,
+    WebGPUCmdUpdateBuffer,
+} from './webgpu-commands';
+
+export class WebGPUCommandPool<T extends WebGPUCmdObject> {
+
+    private _frees: (T|null)[];
+    private _freeIdx: number = 0;
+    private _freeCmds: CachedArray<T>;
+
+    constructor (clazz: new() => T, count: number) {
+        this._frees = new Array(count);
+        this._freeCmds = new CachedArray(count);
+        for (let i = 0; i < count; ++i) {
+            this._frees[i] = new clazz();
+        }
+        this._freeIdx = count - 1;
+    }
+
+    /*
+    public alloc (clazz: new() => T): T {
+        return new clazz();
+    }
+    */
```

Comment:
Need cleanup

Reply:
WebGPUCommandAllocator's clearCmds function is used to make calls.

Reply:
I don't understand. I mean the following unused code can be removed.
```
   /*
    public alloc (clazz: new() => T): T {
        return new clazz();
    }
    */
```